helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -568,7 +568,8 @@ public T get(String path, Stat stat, int options) {
       }
       return records;
     } catch (Exception e) {
-      throw new HelixMetaDataAccessException(String.format("Fail to read nodes for %s", paths));
+      throw new HelixMetaDataAccessException(
+          String.format("Fail to read nodes for %s, err: %s", paths, e.getMessage()));
```

Comment:
How about including the whole exception as the cause for `HelixMetaDataAccessException`?

Reply:
TFTR. Updated.